code/game/objects/items/food/pastries.dm

```diff
@@ -0,0 +1,779 @@
+	foodtypes = GRAIN | SUGAR
+
+#define PANCAKE_MAX_STACK 10
+
+/obj/item/food/pancakes
+	name = "pancake"
+	desc = "A fluffy pancake. The softer, superior relative of the waffle."
+	icon_state = "pancakes_1"
+	inhand_icon_state = "pancakes"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 4, /datum/reagent/consumable/nutriment/vitamin = 2)
+	tastes = list("pancakes" = 1)
+	foodtypes = GRAIN | SUGAR | BREAKFAST
+
+/obj/item/food/pancakes/blueberry
+	name = "blueberry pancake"
+	desc = "A fluffy and delicious blueberry pancake."
+	icon_state = "bbpancakes_1"
+	inhand_icon_state = "bbpancakes"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("pancakes" = 1, "blueberries" = 1)
+
+/obj/item/food/pancakes/chocolatechip
+	name = "chocolate chip pancake"
+	desc = "A fluffy and delicious chocolate chip pancake."
+	icon_state = "ccpancakes_1"
+	inhand_icon_state = "ccpancakes"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("pancakes" = 1, "chocolate" = 1)
+
+/obj/item/food/pancakes/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/item/food/pancakes/update_icon()
+	if(contents.len)
+		name = "stack of pancakes"
+	else
+		name = initial(name)
+	if(contents.len < LAZYLEN(overlays))
+		overlays-=overlays[overlays.len]
+
+/obj/item/food/pancakes/examine(mob/user)
+	var/ingredients_listed = ""
+	var/pancakeCount = contents.len
+	switch(pancakeCount)
+		if(0)
+			desc = initial(desc)
+		if(1 to 2)
+			desc = "A stack of fluffy pancakes."
+		if(3 to 6)
+			desc = "A fat stack of fluffy pancakes!"
+		if(7 to 9)
+			desc = "A grand tower of fluffy, delicious pancakes!"
+		if(PANCAKE_MAX_STACK to INFINITY)
+			desc = "A massive towering spire of fluffy, delicious pancakes. It looks like it could tumble over!"
+	. = ..()
+	if (pancakeCount)
+		for(var/obj/item/food/pancakes/ING in contents)
+			ingredients_listed += "[ING.name], "
+		. += "It contains [contents.len?"[ingredients_listed]":"no ingredient, "]on top of a [initial(name)]."
+
+/obj/item/food/pancakes/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/food/pancakes/))
+		var/obj/item/food/pancakes/P = I
+		if((contents.len >= PANCAKE_MAX_STACK) || ((P.contents.len + contents.len) > PANCAKE_MAX_STACK))
+			to_chat(user, "<span class='warning'>You can't add that many pancakes to [src]!</span>")
+		else
+			if(!user.transferItemToLoc(I, src))
+				return
+			to_chat(user, "<span class='notice'>You add the [I] to the [name].</span>")
+			P.name = initial(P.name)
+			contents += P
+			update_snack_overlays(P)
+			if (P.contents.len)
+				for(var/V in P.contents)
+					P = V
+					P.name = initial(P.name)
+					contents += P
+					update_snack_overlays(P)
+			P = I
+			P.contents.Cut()
+		return
+	else if(contents.len)
+		var/obj/O = contents[contents.len]
+		return O.attackby(I, user, params)
+	..()
```

Comment:
All these variable names are pretty bad, we're trying to avoid one letter variable names now--replace them with variants like `pancakes` etc.

Reply:
this is not really an issue with this PR, it's an issue with the original code, which we didn't touch, we just moved it.